libgadget/init.c

```diff
@@ -210,6 +216,12 @@ inttime_t init(int RestartSnapNum, DomainDecomp * ddecomp)
             if(P[i].Hsml == 0)
                 P[i].Hsml = 0.1 * All.MeanSeparation[0];
         }
+        
+        if(All.FdmOn && RestartSnapNum == -1 && P[i].Type==1)
```

Comment:
Why don't you make the FDM type 3? That way we can run simulations with both CDM and FDM, which is a scientifically interesting case.

Reply:
Hah, I was waiting for you to say that. Will do it.

Reply:
I am predictable :)